src/util/mogps.jl

```diff
@@ -1,3 +1,36 @@
+"""
+    MOutput(x::AbstractVector, out_dim::Integer)
+
+A data type to handle multi-dimensional outputs.
+"""
+struct MOutput{T<:AbstractVector} <: AbstractVector{Real}
+    x::T
+    out_dim::Integer
```

Comment:
Should avoid abstract fields:
```suggestion
    out_dim::Int
```
However, I'm wondering we actually need this - couldn't one just use `length(first(x))` in all places where it's needed?

Reply:
We definitely can. I was unsure show expensive computing `length(first(x))` everytime would be.

Reply:
@devmotion @theogf Do you suggest I remove it?